addons/link_tracker/migrations/13.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,24 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_xmlid_renames = [
+    # ir.actions.act_window
+    ('link_tracker.action_link_tracker', 'link_tracker.link_tracker_action'),
+    ('link_tracker.action_link_tracker_stats', 'link_tracker.link_tracker_action_campaign'),
+    ('link_tracker.action_view_click_statistics', 'link_tracker.link_tracker_click_action_statistics'),
+    # ir.ui.menu
+    ('link_tracker.menu_url_shortener_main', 'link_tracker.link_tracker_menu_main'),
+    # ir.ui.view
+    ('link_tracker.view_link_tracker_click_form', 'link_tracker.link_tracker_click_view_form'),
+    ('link_tracker.view_link_tracker_click_graph', 'link_tracker.link_tracker_click_view_graph'),
+    ('link_tracker.view_link_tracker_click_tree', 'link_tracker.link_tracker_click_view_tree'),
+    ('link_tracker.view_link_tracker_filter', 'link_tracker.link_tracker_view_search'),
+    ('link_tracker.view_link_tracker_form', 'link_tracker.link_tracker_view_form'),
+    ('link_tracker.view_link_tracker_graph', 'link_tracker.link_tracker_view_graph'),
+    ('link_tracker.view_link_tracker_tree', 'link_tracker.link_tracker_view_tree'),
+]
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    openupgrade.rename_xmlids(env.cr, _xmlid_renames)
```

Comment:
no need: act_windows, menus, and views, when they are noupdate=0, then the deprecated ones will be deleted and the new ones will be created.

Reply:
I thought it's better practice if I resolve all xmls deleted or renamed!
If not I'll skip noupdates of act_windows, menus and views

Reply:
Don't waste time, lines of code and so on for something that is done automatically. Here explicit is not better than implicit.